assets/src/edit-story/components/inspector/prepublish/checklistTab.js

```diff
@@ -0,0 +1,199 @@
+            ? ['highPriority', 'recommended']
+            : ['recommended', 'highPriority'];
+
+          if (currentMessage.page !== undefined) {
+            const prevTypeGroup = prevMessages.pages[typeKey] || {};
+            const prevPageGroup = prevTypeGroup[currentMessage.page] || [];
+
+            let prevLengths = { highPriority: 0, recommended: 0 };
+            if (prevMessages.pages?.lengths) {
+              ({ lengths: prevLengths } = prevMessages.pages);
+            }
+
+            return {
+              recommended: prevMessages.recommended,
+              highPriority: prevMessages.highPriority,
+              pages: {
+                ...prevMessages.pages,
+                [typeKey]: {
+                  ...prevTypeGroup,
+                  [currentMessage.page]: [...prevPageGroup, currentMessage],
+                },
+                lengths: {
+                  ...prevLengths,
+                  [typeKey]: prevLengths[typeKey] + 1,
+                },
+              },
+            };
+          }
+
+          const groupedMessages = {
+            [copyKey]: prevMessages[copyKey],
+            [typeKey]: [...prevMessages[typeKey], currentMessage],
+          };
+          return { pages: prevMessages.pages, ...groupedMessages };
+        },
+        {
+          highPriority: [],
+          recommended: [],
+          pages: {},
+        }
+      ),
+    [checklist]
+  );
+
+  const renderRow = useCallback(({ message, id }) => {
+    return <Row key={`guidance-${id}`}>{message}</Row>;
+  }, []);
+
+  const renderPageGroupedRow = (entry) => {
+    const [pageNum, messages] = entry;
+    return (
+      <Fragment key={`guidance-page-group-${pageNum}`}>
+        <PageIndicator>
+          <Rectangle />
+          {sprintf(
+            /* translators: %s: the page number where the checklist issue is. */
+            __('Page %s', 'web-stories'),
+            pageNum
+          )}
+        </PageIndicator>
+        {messages.map(renderRow)}
+      </Fragment>
+    );
+  };
+
+  const showHighPriorityItems =
+    Boolean(highPriority.length) || Boolean(pages.lengths?.highPriority);
+  const showRecommendedItems =
+    Boolean(recommended.length) || Boolean(pages.lengths?.recommended);
+
+  return (
+    <>
+      {showHighPriorityItems && (
+        <SimplePanel
+          name="checklist"
+          title={
+            <>
+              <NumberDecorator error>{highPriority.length}</NumberDecorator>
+              <PanelTitle error>
+                {__('High Priority', 'web-stories')}
+              </PanelTitle>
+            </>
+          }
+        >
+          {highPriority.map(renderRow)}
+          {Object.entries(pages.highPriority || {}).map(renderPageGroupedRow)}
+        </SimplePanel>
+      )}
+      {showRecommendedItems && (
+        <SimplePanel
+          name="checklist"
+          title={
+            <>
+              <NumberDecorator recommended>
+                {recommended.length}
```

Comment:
This number is showing up as 0 for me even though there are lots of items on the story I'm testing.

Reply:
forgot to add the lengths for the ones grouped by their page